### PR TITLE
Introduce Infinispan as cache and hibernate back-end

### PR DESCRIPTION
**Pull request by carlosmunoz**
Make infinispan the backend used for all caches (hibernate included) as well as hibernate search indexes. All configuration for caching now falls on standalone.xml. Additionally, these caches can be made to persist to the database meaning it's now possible to have caches that persist server shutdowns (statistics come to mind here), as well as keep them solely in memory (to test hibernate search without needing to create index files)

### CODE REVIEW

**Comment by carlosmunoz** [comment]
@seanf Can we consider merging this into master now that we've branched for release?

**Comment by carlosmunoz** [comment]
@seanf bump

**Comment by seanf** [comment]
@carlosmunoz Looks good, but we might need QA to look again, after the branch is rebased (or master is merged in).

**Comment by carlosmunoz** [comment]
@zanata-jenkins retest this please

**Comment by carlosmunoz** [comment]
@seanf Care to review the last commits for this pull request?

File: docs/release-notes.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+## 3.7
+
+<h5>Infrastructure Changes</h5>
+
+Zanata now uses Infinispan as its cache provider, and the cache needs to be configured in Jboss' `standalone.xml` file. Please see the [Infinispan](configuration/infinispan) section for more information.
+
```
Comment (seanf) [inline docs/release-notes.md]: I don't think this pull request updates all SIX! copies of standalone*.xml in the source tree.

**Comment by seanf** [comment]
Otherwise looks fine.

File: docs/release-notes.md (hunk 1 above)
Comment (carlosmunoz) [inline docs/release-notes.md]: Added the configuration to all our standalone.xml files.

**Comment by carlosmunoz** [comment]
No. We left hibernate search out of infinispan for the time being due to less than desirable performance.

**Comment by seanf** [comment]
:+1: Once the tests pass.  I would suggest running all functional tests before merging though.

**Pull request merged**